### PR TITLE
Update updateHostsFile.py

### PR DESCRIPTION
**Pull request by iDigitalFlame**
Fixes an issue with reading hosts files that contain non ASCII characters.

### CODE REVIEW

File: updateHostsFile.py [hunk 1]
```diff
@@ -630,8 +630,8 @@ def create_initial_file():
     # spin the sources for the base file
     for source in recursive_glob(settings["datapath"],
                                  settings["hostfilename"]):
-        with open(source, "r") as curFile:
-            write_data(merge_file, curFile.read())
+        with open(source, "rb") as curFile:
```
Comment (gfyoung) [inline updateHostsFile.py]: If this is indeed more robust (testing this is tricky admittedly), we should just open everything as `rb` (i.e. fix all of the cases).  Maybe we go as far as creating an entirely new method.

Comment (FadeMind) [reply updateHostsFile.py]: This patch did not work for me:
```
[fademind@manjaro hosts]$ python3 updateHostsFile.py -a 
Updating source data/add.Risk from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Risk/hosts
Traceback (most recent call last):
  File "updateHostsFile.py", line 1369, in <module>
    main()
  File "updateHostsFile.py", line 161, in main
    update_all_sources(source_data_filename, settings["hostfilename"])
  File "updateHostsFile.py", line 591, in update_all_sources
    update_data = json.load(update_file)
  File "/usr/lib/python3.6/json/__init__.py", line 296, in load
    return loads(fp.read(),
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 27: ordinal not in range(128)
[fademind@manjaro hosts]$ 
```

When I change all `"r" to "rb"` (8 lines) and 3 lines `write_data(merge_file, curFile.read)` into
```
write_data(merge_file, curFile.read().decode('UTF-8'))
```
then:
```
[fademind@manjaro hosts]$ python3 updateHostsFile.py -a 
Updating source data/add.Risk from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Risk/hosts
Updating source data/someonewhocares.org from http://someonewhocares.org/hosts/zero/hosts
Updating source data/adaway.org from https://raw.githubusercontent.com/AdAway/adaway.github.io/master/hosts.txt
Updating source data/tyzbit from https://raw.githubusercontent.com/tyzbit/hosts/master/data/tyzbit/hosts
Updating source data/StevenBlack from https://raw.githubusercontent.com/StevenBlack/hosts/master/data/StevenBlack/hosts
Updating source data/add.2o7Net from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.2o7Net/hosts
Updating source data/UncheckyAds from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/UncheckyAds/hosts
Updating source data/mvps.org from http://winhelp2002.mvps.org/hosts.txt
Updating source data/add.Dead from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Dead/hosts
Updating source data/yoyo.org from https://pgl.yoyo.org/adservers/serverlist.php?hostformat=hosts&mimetype=plaintext&useip=0.0.0.0
Updating source data/KADhosts from https://raw.githubusercontent.com/azet12/KADhosts/master/KADhosts.txt
Updating source data/Badd-Boyz-Hosts from https://raw.githubusercontent.com/mitchellkrogza/Badd-Boyz-Hosts/master/hosts
Updating source data/malwaredomainlist.com from http://www.malwaredomainlist.com/hostslist/hosts.txt
Updating source data/Spotify-Ad-free from https://raw.githubusercontent.com/CHEF-KOCH/Spotify-Ad-free/master/Spotifynulled.txt
Updating source data/add.Spam from https://raw.githubusercontent.com/FadeMind/hosts.extras/master/add.Spam/hosts
Updating source extensions/fakenews from https://raw.githubusercontent.com/marktron/fakenews/master/fakenews
Updating source extensions/social from https://raw.githubusercontent.com/Sinfonietta/hostfiles/master/social-hosts
Updating source extensions/porn/clefspeare13 from https://raw.githubusercontent.com/Clefspeare13/pornhosts/master/0.0.0.0/hosts
Updating source extensions/porn/sinfonietta-snuff from https://raw.githubusercontent.com/Sinfonietta/hostfiles/master/snuff-hosts
Updating source extensions/porn/sinfonietta from https://raw.githubusercontent.com/Sinfonietta/hostfiles/master/pornography-hosts
Updating source extensions/gambling from https://raw.githubusercontent.com/Sinfonietta/hostfiles/master/gambling-hosts
Traceback (most recent call last):
  File "updateHostsFile.py", line 1369, in <module>
    main()
  File "updateHostsFile.py", line 190, in main
    final_file = remove_dups_and_excl(merge_file, exclusion_regexes)
  File "updateHostsFile.py", line 711, in remove_dups_and_excl
    line = line.strip(" \t\n\r")
TypeError: a bytes-like object is required, not 'str'
[fademind@manjaro hosts]$ 
```

cutoff

```
Traceback (most recent call last):
  File "updateHostsFile.py", line 1369, in <module>
    main()
  File "updateHostsFile.py", line 190, in main
    final_file = remove_dups_and_excl(merge_file, exclusion_regexes)
  File "updateHostsFile.py", line 711, in remove_dups_and_excl
    line = line.strip(" \t\n\r")
TypeError: a bytes-like object is required, not 'str'
```

190 LINE **final_file = remove_dups_and_excl(merge_file, exclusion_regexes)**
```
if settings["compress"]:
        # Another mode is required to read and write the file in Python 3
        final_file = open(path_join_robust(settings["outputpath"], "hosts"),
                          "w+b" if PY3 else "w+")
        compressed_file = tempfile.NamedTemporaryFile()
        remove_dups_and_excl(merge_file, exclusion_regexes, compressed_file)
        compress_file(compressed_file, settings["targetip"], final_file)
    else:
        final_file = remove_dups_and_excl(merge_file, exclusion_regexes)
```
LINE 709 **with open(settings["whitelistfile"], "rb") as ins:**
```
def remove_dups_and_excl(merge_file, exclusion_regexes, output_file=None):
    """
    Remove duplicates and remove hosts that we are excluding.

    We check for duplicate hostnames as well as remove any hostnames that
    have been explicitly excluded by the user.

    Parameters
    ----------
    merge_file : file
        The file object that contains the hostnames that we are pruning.
    exclusion_regexes : list
        The list of regex patterns used to exclude domains.
    output_file : file
        The file object in which the result is written. If None, the file
        'settings["outputpath"]' will be created.
    """

    number_of_rules = settings["numberofrules"]
    if os.path.isfile(settings["whitelistfile"]):
        with open(settings["whitelistfile"], "rb") as ins:
            for line in ins:
                line = line.strip(" \t\n\r")
                if line and not line.startswith("#"):
                    settings["exclusions"].append(line)

    if not os.path.exists(settings["outputpath"]):
        os.makedirs(settings["outputpath"])

    if output_file is None:
        # Another mode is required to read and write the file in Python 3
        final_file = open(path_join_robust(settings["outputpath"], "hosts"),
                          "w+b" if PY3 else "w+")
    else:
        final_file = output_file
```

Comment (gfyoung) [reply updateHostsFile.py]: I would suspect then that you have to modify the `.strip` class so that we aren't handling raw strings.

**Comment by FadeMind** [comment]
@StevenBlack  this PR is useless now.

**Comment by StevenBlack** [comment]
Thanks Tomasz @FadeMind.  I wouldn't say *useless* 😄, obsolete now, yes.  We're merging #520 soon, so this can be closed.  Thanks @iDigitalFlame!

**Comment by FadeMind** [comment]
Corrected 😄

**Pull request closed without merging**